src/octopeer-github/test/ElementSelectionBehaviour/SelectButtonsElementSelectionBehaviourTest.ts

```diff
@@ -1,26 +1,153 @@
+/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/LockConversationSettingElementSelectionBehaviour.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/MilestoneSettingElementSelectionBehaviour.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/SettingElementSelectionBehaviour/UnsubscribeSettingElementSelectionBehaviour.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/CommitsTabHeaderElementSelectionBehaviour.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/ConversationTabHeaderElementSelectionBehaviour.ts"/>
+/// <reference path="../../content/ElementSelectionBehaviour/TabHeaderElementSelectionBehaviour/FilesChangedTabHeaderElementSelectionBehaviour.ts"/>
+/* tslint:enable */
 
-describe("An ElementSelector that selects Buttons", function() {
-    let selector: ElementSelectionBehaviour;
-    let database: DatabaseAdaptable;
-    let logSpy: jasmine.Spy;
-
-    beforeEach(function() {
-        setFixtures("<div><button id='bt1' class='js-merge-branch-action'></button><button id='bt2' class='btn2'></button></div>");
-        database = new ConsoleLogDatabaseAdapter();
-        selector = new MergePRButtonElementSelectionBehaviour(database);
-        logSpy = spyOn(database, "post");
-    });
-
-    it("should select buttons", function () {
-        expect(selector.getElements().length).toBe(1);
-    });
-
-    it("has a callback that should post to the database", function () {
-        const mockedClick = $.Event("click");
-        selector.getCallback(new EventID(1))(mockedClick);
-        expect(logSpy).toHaveBeenCalled();
-    });
-});
+interface CreatableNumberPair extends Array<ElementSelectionBehaviourCreatable | number> {
+    0: ElementSelectionBehaviourCreatable;
+    1: number;
+}
+
+// TODO: All commented tuples in the lists have to be fixed with Ajax request first
+let conversationSelectors: CreatableNumberPair[] =  [
+    [AddEmoticonButtonElementSelectionBehaviour, 2],
+    [CancelEditPRNameButtonElementSelectionBehaviour, 0],
+    [CheckDetailButtonElementSelectionBehaviour, 2],
+    [ClosePRButtonElementSelectionBehaviour, 1],
+    [CommentPRButtonElementSelectionBehaviour, 1],
+    [EditCommentButtonElementSelectionBehaviour, 2],
+    [EditPRNameButtonElementSelectionBehaviour, 1],
+    [MergePRButtonElementSelectionBehaviour, 1],
+    [SaveEditPRNameButtonElementSelectionBehaviour, 0],
+    [ShowChecksToggleButtonElementSelectionBehaviour, 1],
+    [DateMiscellaneousElementSelectionBehaviour, 8],
+    [CommitHashcodeNameElementSelectionBehaviour, 4],
+    [CommitMessageNameElementSelectionBehaviour, 4],
+    // [OtherContributerNameElementSelectionBehaviour, 0],
+    // [PRCreatorNameElementSelectionBehaviour, 3],
+    // [PRParticipantNameElementSelectionBehaviour, 4],
```

Comment:
I'm assuming these lines are commented because the NameSelectors don't have a proper selector yet? (maybe add a todo so that it's clear for everyone :) )